### PR TITLE
Adding Tests for ExactMatch, IntegerMatch and StringMatch

### PR DESCRIPTION
**Pull request by mikaelm12**
Added tests for ExactMatch, IntegerMatch and StringMatch in the Rewrite project
@Tratcher @natemcmaster

### CODE REVIEW

File: test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/ExactMatchTests.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+namespace Microsoft.AspNetCore.Rewrite.Tests.UrlMatches
+{
+    public class ExactMatchTests
+    {
+        [Theory]
+        [InlineDataAttribute(true,"string",false,"string",true)]
+        [InlineDataAttribute(false, "STRING", false, "string",false)]
+        [InlineDataAttribute(true, "string", true, "string", false)]
+        public void ExactMatch_Case_Sensitivity_Negate_Tests(bool ignoreCase,string inputString,bool negate,string pattern,bool expectedResult)
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/ExactMatchTests.cs]: Nit: format document to add spacing between parameters

File: test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/ExactMatchTests.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+using Microsoft.AspNetCore.Rewrite.Internal.UrlActions;
+using Microsoft.AspNetCore.Rewrite.Internal.UrlMatches;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Rewrite.Tests.UrlMatches
+{
+    public class ExactMatchTests
+    {
+        [Theory]
+        [InlineDataAttribute(true,"string",false,"string",true)]
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/ExactMatchTests.cs]: There should be _at least_ 4 cases to get all combinations of bool values

File: test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/IntegerMatchTests.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Rewrite;
+using Microsoft.AspNetCore.Rewrite.Internal;
+using Microsoft.AspNetCore.Rewrite.Internal.UrlActions;
+using Microsoft.AspNetCore.Rewrite.Internal.UrlMatches;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Rewrite.Tests.UrlMatches
+{
+    public class IntegerMatchTests
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/IntegerMatchTests.cs]: Other tests that might be good to add:
"0", string.Empty, integers to big for the range of `System.Int32`

File: test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/StringMatchTests.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+{
+    public class StringMatchTests
+    {
+        [Theory]
+        [InlineData("hi", StringOperationType.Equal,true,"hi",true)]
+        [InlineData("a", StringOperationType.Greater, true, "b", true)]
+        [InlineData("a", StringOperationType.GreaterEqual, true, "b", true)]
+        [InlineData("b", StringOperationType.Less,true, "a", true)]
+        [InlineData("b", StringOperationType.LessEqual, true, "a", true)]
+        public void StringMatch_Evaluation_Check_Cases(string value, StringOperationType operation, bool ignoreCase, string input, bool expectedResult)
```
Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Rewrite.Tests/UrlMatches/StringMatchTests.cs]: Other tests to add: comparisons with string.Empty and null

**Comment by natemcmaster** [comment]
:watch: (will review more after responses to PR feedback)

File: test/Microsoft.AspNetCore.Rewrite.Tests/project.json [hunk 5]
```diff
@@ -12,7 +13,9 @@
   },
   "frameworks": {
     "netcoreapp1.0": {
+      "imports": "dnxcore50",
       "dependencies": {
+        "Microsoft.CodeCoverage": "1.0.1",
```
Comment (Tratcher) [inline test/Microsoft.AspNetCore.Rewrite.Tests/project.json]: Either remove this before checking in, or make it a build time dependency

Comment (natemcmaster) [inline test/Microsoft.AspNetCore.Rewrite.Tests/project.json]: i.e.
```json
 "Microsoft.CodeCoverage": {
     "type": "build",
     "version": "1.0.1"
  },
```

**Comment by natemcmaster** [comment]
:shipit:

**Pull request closed without merging**